### DISCUSSION TITLE
Welcome to Computational Physics Course Discussions page!

### DISCUSSION THREAD

**Discussion by williamgilpin** [williamgilpin/cphy#1] [Announcements]
## 👋 Welcome!

I want to try out using GitHub discussions instead of Slack , Discord, or Piazza, to see if the interface is useful. Feel free to test posts on this thread. You can tag specific parts of the host repository, etc. If you want to edit the course content, or find a typo, please open a traditional GitHub issue or submit a pull request.

**Comment by adamscottc** [comment]
I'm new to using Python outside of a web environment so the process of downloading the different programs and setting everything up took some time. If anyone is struggling to get things to work, I used Anaconda as my package and environment manager and Visual Studio Code as my source code editor. 
Cons: There always seemed to be another program or package to install, or an environment to activate before I could officially. For someone who knew more about what they needed, a brief internet search could have solved those problems fairly simply, but since I didn't know what terms to use.
Pros: Once you know you're way around, those programs are intuitive and quick to use.

**Reply by williamgilpin** [reply]
yeah dependency management in Python can get really brutal... for some deep learning libraries your deps get coupled to your hardware because of the GPU drivers (CUDA, etc) and it gets really bad.

conda should help some, since it insulates different projects from one another.

if there's anything in particular that can improve the handout (or useful links, particular gotchas etc) feel free to PR the course repo or post more
